injected/src/features/message-bridge/create-page-world-bridge.js

```diff
@@ -9,126 +10,146 @@ import {
 } from './schema.js';
 
 /**
- * todo: use the already-captured ones
+ * @import { Messaging } from "@duckduckgo/messaging";
+ * @typedef {Pick<Messaging, 'notify' | 'request' | 'subscribe'>} MessagingInterface
+ * @typedef {Pick<import("../../captured-globals.js"),
+ *    "dispatchEvent" | "addEventListener" | "removeEventListener" | "CustomEvent" | "String" | "mathRandom">
+ * } Captured
  */
-const captured = {
-    dispatchEvent: globalThis.dispatchEvent.bind(globalThis),
-    addEventListener: globalThis.addEventListener.bind(globalThis),
-    removeEventListener: globalThis.removeEventListener.bind(globalThis),
-    CustomEvent: globalThis.CustomEvent,
-    Promise: globalThis.Promise,
-    String: globalThis.String,
-    mathRandom: globalThis.Math.random.bind(globalThis.Math),
-};
+/** @type {Captured} */
+const captured = capturedGlobals;
 
 /**
  * This part can be called from any script.
  * @param {string} featureName
- * @param {string} token
- * @return {Pick<import("@duckduckgo/messaging").Messaging, 'notify' | 'subscribe' | 'request'>}
+ * @param {string} [token]
+ * @return {MessagingInterface}
  */
 export function createPageWorldBridge(featureName, token) {
-    console.log('will createPageWorldBridge for', featureName, 'with token', token);
-
-    /**
-     * TODO: what is the value of this token? - since 'createPageWorldBridge' is available to all page scripts, probably none
-     * @param {string} eventName
-     * @return {`${string}-${string}`}
-     */
-    function appendToken(eventName) {
+    const appendToken = (eventName) => {
         return `${eventName}-${token}`;
-    }
+    };
 
     /**
+     * Create the sender to centralize the sending logic
      * @param {{name: string} & Record<string, any>} incoming
      */
-    function send(incoming) {
+    const send = (incoming) => {
+        // when the token is absent, just silently fail
+        if (!token) return;
         const event = new captured.CustomEvent(appendToken(incoming.name), { detail: incoming });
         captured.dispatchEvent(event);
-    }
+    };
+
+    send(new InstallProxy({ featureName: this.featureName }));
 
-    function random() {
-        return captured.String(captured.mathRandom());
+    return new PageWorldMessagingBridge({ featureName, send, appendToken });
+}
+
+/**
+ * This is what is returned to consumers - just `notify`, `request` and `subscribe`
+ * @implements {MessagingInterface}
+ */
+class PageWorldMessagingBridge {
+    random = random;
+    /**
+     * @param {object} params
+     * @param {string} params.featureName
+     * @param {(evt: {name: string} & Record<string, any>) => void} params.send
+     * @param {(s: string) => string} params.appendToken
+     */
+    constructor({ featureName, send, appendToken }) {
+        this.featureName = featureName;
+        this.send = send;
+        this.appendToken = appendToken;
```

Comment:
hm, should we not expose this to the consumers?

Reply:
Yeah good point actually - I'll move to function + closure to hide stuff